tests/python/test-5-operator.py

```diff
@@ -1551,6 +1551,342 @@ def test_533(ceed_resource):
+                                   cmode=libceed.USE_POINTER)
+    strides = np.array([1, q, q], dtype="int32")
+
+    rui = ceed.StridedElemRestriction(nelem, q, 1, q * nelem, strides)
+
+    # Bases
+    bx = ceed.BasisTensorH1Lagrange(1, 1, 2, q, libceed.GAUSS)
+    bu_fine = ceed.BasisTensorH1Lagrange(1, ncomp, p_fine, q, libceed.GAUSS)
+
+    # QFunctions
+    file_dir = os.path.dirname(os.path.abspath(__file__))
+    qfs = load_qfs_so()
+
+    qf_setup = ceed.QFunction(1, qfs.setup_mass,
+                              os.path.join(file_dir, "test-qfunctions.h:setup_mass"))
+    qf_setup.add_input("weights", 1, libceed.EVAL_WEIGHT)
+    qf_setup.add_input("dx", 1, libceed.EVAL_GRAD)
+    qf_setup.add_output("rho", 1, libceed.EVAL_NONE)
+
+    qf_mass = ceed.QFunction(1, qfs.apply_mass_two,
+                             os.path.join(file_dir, "test-qfunctions.h:apply_mass_two"))
+    qf_mass.add_input("rho", 1, libceed.EVAL_NONE)
+    qf_mass.add_input("u", ncomp, libceed.EVAL_INTERP)
+    qf_mass.add_output("v", ncomp, libceed.EVAL_INTERP)
+
+    # Operators
+    op_setup = ceed.Operator(qf_setup)
+    op_setup.set_field("weights", libceed.ELEMRESTRICTION_NONE, bx,
+                       libceed.VECTOR_NONE)
+    op_setup.set_field("dx", rx, bx, libceed.VECTOR_ACTIVE)
+    op_setup.set_field("rho", rui, libceed.BASIS_COLLOCATED,
+                       libceed.VECTOR_ACTIVE)
+
+    op_mass_fine = ceed.Operator(qf_mass)
+    op_mass_fine.set_field("rho", rui, libceed.BASIS_COLLOCATED, qdata)
+    op_mass_fine.set_field("u", ru_fine, bu_fine, libceed.VECTOR_ACTIVE)
+    op_mass_fine.set_field("v", ru_fine, bu_fine, libceed.VECTOR_ACTIVE)
+
+    # Setup
+    op_setup.apply(x, qdata)
+
+    # Create multigrid level
+    p_mult_fine = ceed.Vector(ncomp * nu_fine)
+    p_mult_fine.set_value(1.0)
+    [op_mass_coarse, op_prolong, op_restrict] = op_mass_fine.multigrid_create_tensor_h1_lagrange(p_mult_fine,
+                                                                                                 ru_coarse, p_coarse)
+
+    # Apply coarse mass matrix
+    u_coarse.set_value(1.0)
+    op_mass_coarse.apply(u_coarse, v_coarse)
+
+    # Check
+    with v_coarse.array_read() as v_array:
+        total = 0.0
+        for i in range(nu_coarse * ncomp):
+            total = total + v_array[i]
+        assert abs(total - 2.0) < 1E-13
+
+    # Prolong coarse u
+    op_prolong.apply(u_coarse, u_fine)
+
+    # Apply mass matrix
+    op_mass_fine.apply(u_fine, v_fine)
+
+    # Check
+    with v_fine.array_read() as v_array:
+        total = 0.0
+        for i in range(nu_fine * ncomp):
+            total = total + v_array[i]
+        assert abs(total - 2.0) < 1E-13
+
+    # Restrict state to coarse grid
+    op_restrict.apply(v_fine, v_coarse)
+
+    # Check
+    with v_coarse.array_read() as v_array:
+        total = 0.0
+        for i in range(nu_coarse * ncomp):
+            total = total + v_array[i]
+        assert abs(total - 2.0) < 1E-13
+
```

Comment:
I'm a bit puzzled... what is the difference between this and the previous test? They have the same name and description

Reply:
Weird, removed

Reply:
No problem! I just wanted to make sure I was not seeing double on my screen! 😅😂